File: 10810.py
N, M = map(int, input().split())
baskets = [0] * (N + 1)

for i in range(M):
    I, J, K = map(int, input().split())

    for j in range(I, J + 1):
        baskets[j] = K

for basket in baskets[1:]:
    print(basket, end=' ')
print()

File: 10810.cpp
#include <iostream>
using namespace std;

int* baskets;

int main() {
	int N, M; cin >> N >> M;
	baskets = new int[N + 1];
	for (int i = 1; i <= N; i++)
		baskets[i] = 0;

	for (int i = 0; i < M; i++) {
		int I, J, K; cin >> I >> J >> K;

		for (int j = I; j <= J; j++)
			baskets[j] = K;
	}

	for (int i = 1; i <= N; i++)
		cout << baskets[i] << ' ';
	cout << endl;
}
